modules/nextflow/src/main/groovy/nextflow/k8s/model/PodSpecBuilder.groovy

```diff
@@ -307,15 +314,19 @@ class PodSpecBuilder {
     }
 
     Map build() {
-        assert this.podName, 'Missing K8s podName parameter'
+        // podname is not set in case of jobs
         assert this.imageName, 'Missing K8s imageName parameter'
         assert this.command || this.args, 'Missing K8s command parameter'
 
         final restart = this.restart ?: 'Never'
 
         final metadata = new LinkedHashMap<String,Object>()
-        metadata.name = podName
-        metadata.namespace = namespace ?: 'default'
+        // if podName -> create pod and need podName and namespace, 
+        //  else creating job without podName and namespace
+        if( this.podName ) {
+            metadata.name = podName
+            metadata.namespace = namespace ?: 'default'
+        }
```

Comment:
What the logic behind this? I mean why for the Job should be different ?

Reply:
You should set namespace and name only to job part of the object, not pod object. For job case, we do not set podName, thus not filling pod metadata for job, these attributes seem to be ignored at all.